modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/revision/APIRevisionTestCase.java

```diff
@@ -526,8 +530,21 @@ public void testInvokeAPIInRetiredLifecycleStage() throws Exception {
                 "Unable to get error for invoking API in RETIRED stage using application subscription token");
     }
 
-    @AfterClass(alwaysRun = true)
-    public void destroy() throws Exception {
+    @Test(groups = {"wso2.am"}, description = "Test traces of the deleted API wont appear in admin console",
+            dependsOnMethods = "testInvokeAPIInRetiredLifecycleStage")
+    public void testIfTracesOfDeletedApisVisible() throws Exception {
+        API_TRACES_LOCATION = API_TRACES_LOCATION.concat(apiId);
+        resourceAdminServiceClient =
+                new ResourceAdminServiceClient(publisherContext.getContextUrls().getBackEndUrl(),
+                        createSession(publisherContext));
+        ResourceData[] apiResourcesTraces = resourceAdminServiceClient.getResourceData(API_TRACES_LOCATION);
+        assertTrue(apiResourcesTraces[0].getName().equals(apiId));
```

Comment:
we can reverse the order here to prevent any null pointer exceptions

Reply:
solved